### ISSUE TITLE
make check fails on Windows

### ISSUE THREAD

**Issue by saraedum** [Normaliz/Normaliz#258]
Running `make check` on Windows fails for me with:
```
make  check-local
make[2]: Entering directory '/d/bld/normaliz_1566256088326/work/test'
rm -rf run_tests
mkdir run_tests
(cd . && tar cf - test-/23.in test-/23.ref test-/3x3_sign.in […] test-Qdegenerate/empty2pointM.in test-Qdegenerate/empty2pointM.ref test-Qdegenerate/point0M.in test-Qdegenerate/point0M.ref test-Qdegenerate/point0V.in test-Qdegenerate/point0V.ref) | (cd run_tests && tar xf -)
(cd run_tests && make -f /d/bld/normaliz_1566256088326/work/test/Makefile.classic all    testAut NORMALIZ=/d/bld/normaliz_1566256088326/work/source/normaliz)
make[3]: *** No rule to make target 'test-s/rafa2310.diff', needed by 'tests'.  Stop.
```

Should the tests pass on Windows?

See https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=63630&view=logs for the full log.

**Comment by w-bruns** [comment]
We have never run the tests under Windows, but the Windows binary can be checked under Linux via test/normaliz-windows.sh by setting NORMALIZ equal to it and then running make -f Makefile.classic. (The script shows where we usually place normaliz.exe before releasing it.)

The final targets of make are the diff files. They depend on the out files and the ref files.

Our binary for Windows does not use any optional library. It would be great if you could link them in. Without the optional libraries the maximal extent of the tests is defined by all.

**Comment by w-bruns** [comment]
For Windows compilation we are using the Intel compiler via Visual Studio installed on a laptop that we would like to stop using. For this reason it would be good to have another way to compiling Normaliz under MS Windows. What are you using?

As far as we know, the MIcrosoft compiler does not support OpenMP > 2.0.

**Comment by videlec** [comment]
It is likely that the problem comes from the handwritten `Makefile_classical` that is here to run the tests. The testing procedure works as follows: you look at various input files (`%.in`). For each of them you run normaliz with a specific set of options and obtain a result file (`%.out`). Then you make a diff (`%.diff`) with the reference file (`%.ref`).
It is not clear to me:
- can this be done without a Makefile
- can this be done in a Windows compatible way

**Comment by videlec** [comment]
Just read a nice sentence from [this blog](http://skramm.blogspot.com/2013/04/writing-portable-makefiles.html)
```
Anyway, the two shells (Linux/bash and Windows/cmd.exe) are sooo different, only insane people would consider trying to write a "compatible" script, running equally on both systems.
```

**Comment by saraedum** [comment]
We agreed to:
* remove `-lpthread` since openmp flags take care of that
* fix the `-fopenmp` problem locally on conda (as the wrapper seems to be to blame.)
* turn the tests into something more standard
* use `--with-*` in the `configure.ac`.

**Comment by w-bruns** [comment]
-lpthread removed in current master

**Comment by isuruf** [comment]
Note that the conda build log linked runs normaliz in a bash shell on windows.

> As far as we know, the MIcrosoft compiler does not support OpenMP > 2.0.

conda recipe uses `clang-cl`, a MSVC ABI compiler from LLVM project. They also support OpenMP 4.5, but not in a compatible way with MSVC's OpenMP.

**Comment by w-bruns** [comment]
This behavior is intended! But perhaps it would bee better to remove these tests.

**Comment by w-bruns** [comment]
Fixed in 3.8.0
